src/Grid.php

```diff
@@ -138,6 +138,67 @@ public function addButton($text)
         return $this->menu->addItem()->add(new Button($text));
     }
 
+    /**
+     * Add Dropdown menu in grid menu in order to dynamically setup number of item per page.
+     *
+     * @param array  $items An array of item's per page value.
+     * @param null   $ipp   The default item's per page value.
+     * @param string $label The memu item label.
+     *
+     * @throws Exception
+     *
+     * @return $this
+     */
+    public function addRowLimiter($items = [10, 25, 50, 100], $ipp = null, $label = 'Items per page')
+    {
+        if (!$this->menu) {
+            throw new Exception(['Unable to add QuickSearch without Menu']);
+        }
+
+        if (!$ipp) {
+            $ipp = $items[0];
+        }
+
+        $this->ipp = $ipp;
+        $limiter = $this->menu->add('View')->addClass('ui dropdown');
+
+        $menuItems = [];
+        foreach ($items as $key => $item) {
+            $menuItems[] = ['name' => $item, 'value' => $item];
+        }
+
+        //Callback later will give us time to properly render menu item before final output.
+        $cb = $this->add(new CallbackLater());
```

Comment:
CallbackLater give us a chance to properly render menu button attach to grid->menu. Otherwise, the grid is reload without the menu. 
Using Callback would force user to set grid menu item prior to use pagePerItem selector.

So
 ```
$g = $app->add(['Grid']);
$g->setModel(new Country($db));
$g->addQuickSearch();
$g->addRowLimiter();

$g->menu->addItem(['Add Country', 'icon' => 'add square'], new \atk4\ui\jsExpression('alert(123)'));
$g->menu->addItem(['Re-Import', 'icon' => 'power'], new \atk4\ui\jsReload($g));
$g->menu->addItem(['Delete All', 'icon' => 'trash', 'red active']);
```

This would not work with Callback but work using CallbackLater.